cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/UrlDownloadUtil.java

```diff
@@ -0,0 +1,38 @@
+package mil.army.usace.hec.cwms.http.client;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
+public class UrlDownloadUtil {
+
+    private UrlDownloadUtil() {
+        throw new AssertionError("Utility class for generating InputStream for given URL");
+    }
+
+    /**
+     * Get Input Stream.
+     *
+     * @param url - url of file to generate input stream of
+     * @return InputStream
+     * @throws IOException - thrown if execution of request fails
+     */
+    public static InputStream getStream(URL url) throws IOException {
+        Response response = getResponse(url);
```

Comment:
This should utilize existing API by adding capabilities to HttpRequestResponse to be able to get you a byteStream

Reply:
Updated